tests/scenarios/compat-stage2-test.ts

```diff
@@ -491,105 +577,148 @@ stage2Scenarios
-          .isTemplateOnlyComponent('./templates/components/first-choice.hbs');
-      });
+          .withContents(contents => {
+            assert.ok(
+              /setComponentTemplate\(template, templateOnlyComponent\(undefined, "third-choice"\)\);/.test(contents)
+            );
+            return true;
+          });
 
-      test('addon/hello-world.js', function () {
-        expectAudit.module('./node_modules/my-addon/components/hello-world.js').codeEquals(`
-        window.define("my-app/components/second-choice", function () {
-          return importSync("#embroider_compat/components/second-choice");
-        });
-        window.define("my-addon/synthetic-import-1", function () {
-          return importSync("../synthetic-import-1");
-        });
-        import Component from '@ember/component';
-        import layout from '../templates/components/hello-world';
-        import computed from '@ember/object/computed';
-        import somethingExternal from 'not-a-resolvable-package';
-        import { importSync } from "@embroider/macros";
-        export default Component.extend({
-          dynamicComponentName: computed('useDynamic', function () {
-            return this.useDynamic || 'default-dynamic';
-          }),
-          layout
-        });
-        `);
+        expectModule
+          .resolves(/\/components\/first-choice.hbs\/-embroider-pair-component/)
+          .toModule()
+          .withContents(contents => {
+            assert.ok(
+              /setComponentTemplate\(template, templateOnlyComponent\(undefined, "first-choice"\)\);/.test(contents)
+            );
+            return true;
+          });
       });
 
       test('app/hello-world.js', function () {
-        expectAudit.module('./node_modules/my-addon/_app_/components/hello-world.js').codeEquals(`
-          window.define("my-addon/synthetic-import-1", function () {
-            return importSync("my-addon/synthetic-import-1");
-          });
-          import { importSync } from '@embroider/macros';
-          export { default } from 'my-addon/components/hello-world';
-        `);
+        // TODO: it's basically but replace codeEqual with more flexible content checks
+        expectAudit
+          .module('./templates/index.hbs')
+          .resolves(/my-addon\/_app_\/components\/hello-world/)
+          .toModule().codeEquals(`
+            window.define("my-addon/synthetic-import-1", function () {
+              return importSync("my-addon/synthetic-import-1");
+            });
+            import { importSync } from '@embroider/macros';
+            export { default } from 'my-addon/components/hello-world';
+          `);
+      });
 
+      test('addon/hello-world.js', function () {
         expectAudit
-          .module('./node_modules/my-addon/_app_/components/hello-world.js')
-          .resolves('my-addon/components/hello-world')
-          .to('./node_modules/my-addon/components/hello-world.js', 'remapped to precise copy of my-addon');
```

Comment:
I removed this assertion because I need to go through this resolve path anyway to test the content of `my-addon/components/hello-world.js` in another test. So I simply moved the message "remapped to precise copy of my-addon" to a comment in that other test.